tests/unit/etl/NFTHelpersTests.cpp

```diff
@@ -17,45 +17,450 @@
 */
 //==============================================================================
 
+#include "data/DBHelpers.hpp"
 #include "etl/NFTHelpers.hpp"
 #include "util/LoggerFixtures.hpp"
 #include "util/TestObject.hpp"
 
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/spawn.hpp>
+#include <boost/json/array.hpp>
+#include <boost/json/object.hpp>
+#include <boost/json/parse.hpp>
+#include <boost/json/value.hpp>
+#include <gmock/gmock.h>
+#include <grpcpp/support/status.h>
 #include <gtest/gtest.h>
+#include <xrpl/basics/Blob.h>
+#include <xrpl/basics/Slice.h>
 #include <xrpl/basics/base_uint.h>
+#include <xrpl/protocol/AccountID.h>
+#include <xrpl/protocol/SField.h>
+#include <xrpl/protocol/STObject.h>
 #include <xrpl/protocol/STTx.h>
 #include <xrpl/protocol/Serializer.h>
+#include <xrpl/protocol/TER.h>
+#include <xrpl/protocol/TxFormats.h>
 #include <xrpl/protocol/TxMeta.h>
+#include <xrpl/protocol/UintTypes.h>
+#include <xrpl/protocol/tokens.h>
 
+#include <optional>
+#include <stdexcept>
 #include <string>
+#include <string_view>
 #include <vector>
 
 constexpr static auto ACCOUNT = "rM2AGCCCRb373FRuD8wHyUwUsh2dV4BW5Q";
+constexpr static auto ACCOUNT2 = "rnd1nHuzceyQDqnLH8urWNr4QBKt4v7WVk";
 constexpr static auto NFTID = "0008013AE1CD8B79A8BCB52335CD40DE97401B2D60A828720000099B00000000";
 constexpr static auto NFTID2 = "05FB0EB4B899F056FA095537C5817163801F544BAFCEA39C995D76DB4D16F9DA";
 constexpr static auto TX = "13F1A95D7AAB7108D5CE7EEAF504B2894B8C674E6D68499076441C4837282BF8";
+// Page index is a valid nft page for ACCOUNT
+constexpr static auto PAGE_INDEX = "E1CD8B79A8BCB52335CD40DE97401B2D60A82872FFFFFFFFFFFFFFFFFFFFFFFF";
+constexpr static auto OFFER_ID = "AA86CBF29770F72FA3FF4A5D9A9FA54D6F399A8E038F72393EF782224865E27F";
 
-struct NFTHelpersTests : public NoLoggerFixture {};
+struct NFTHelpersTest : NoLoggerFixture {};
 
-TEST_F(NFTHelpersTests, ConvertDataFromNFTCancelOfferTx)
+void
+VerifyNFTTransactionsData(
```

Comment:
It is better to put such functions into fixture to avoid name collision because all tests are linked into one binary. Also  function name should start by a lowercase letter.